cli/testflinger_cli/__init__.py

```diff
@@ -400,6 +427,66 @@ def status(self):
                 "connection or try again later."
             )
 
+    def agent_status(self):
+        """Show the status of a specified agent."""
+        agent_status = self.get_agent_state(self.args.agent_name)
+        if agent_status == "unknown":
+            print(
+                "Unable to retrieve agent state from the server, check your "
+                "connection or try again later."
+            )
+            return
+
+        if self.args.json:
+            output = json.dumps(
+                {
+                    "agent": self.args.agent_name,
+                    "status": agent_status["state"],
+                    "queues": agent_status["queues"],
+                }
+            )
+        else:
+            output = agent_status["state"]
+        print(output)
+
+    def queue_status(self):
+        """Show the status of the agents in a specified queue."""
+        queue_state = self.get_queue_state(self.args.queue_name)
+        jobs_queued = self.get_queued_jobs(self.args.queue_name)
+        if queue_state == "unknown":
+            print(
+                "Unable to retrieve queue state from the server, check your "
+                "connection or try again later."
+            )
+            return
+
+        if self.args.json:
+            output = json.dumps(
+                {
+                    "queue": self.args.queue_name,
+                    "jobs waiting": jobs_queued,
+                    "agents": queue_state,
+                }
+            )
+        else:
+            agents = Counter()
+            for agent in queue_state:
+                status = (
+                    agent["state"]
+                    if agent["state"] in ("waiting", "offline")
+                    else "busy"
+                )
+                agents[status] += 1
+
```

Comment:
I think this can be simplified

```suggestion
            agents = Counter(
                agent["state"]
                if agent["state"] in ("waiting", "offline")
                else "busy"
            )
```